### ISSUE TITLE
Reconcile & Account Balance When Income In Future

### ISSUE THREAD

**Issue by RichInSQL** [actualbudget/releases#209]
When an income item is in the register with a date in the future reconciling the account includes this income even though the date has not yet passed. 

The account balance also shows that the income value that was entered in the register is in the account balance - should it not be included in the register and added to the balance and available for reconcile once the date of the transaction is true?

**Comment by Deguello** [comment]
I haven't used the Reconcile functionality because I'm not sure what the purpose would be without a reconciled flag but it kind of doesn't make sense to reconcile something that hasn't happened yet, so it does seem like future transactions wouldn't be relevant.

That said, for account balance, I kind of think I'd want my account balance on the left to match my running balance in that account. Having the running balance say one think and the account on the left say another would seem odd to me. Additionally, if I have an automatic payment (or whatever) due in three days and I post that transaction, in my view that money is gone and my balance should reflect that.

**Comment by MikeBishop** [comment]
I'm in the (possibly small?) group of people who prefer's nYNAB's behavior of keeping future-dated expenses separate from present/past expenses.  I've entered something because it's going to happen, but it hasn't yet or it wouldn't be dated in the future.

**Comment by Deguello** [comment]
>  I've entered something because it's going to happen, but it hasn't yet or it wouldn't be dated in the future.

I've entered something because it's going to happen and I want it to impact my budget or I wouldn't have entered it yet. When I get a statement notification, I enter that amount using the due date. That transaction is going to happen. It's going to affect my category and account balances. If, for whatever reason, I didn't want it to, I wouldn't have entered it.

> if I have a transaction with a future date that should leave my budget but not my balance as it is still in my account

If the payment scheduled for yesterday takes three days to clear it's still in your account, too. It's easy enough to view the cleared balance in the register, but it would feel really broken to me if the account bar showed a balance different from the final running total in the register.

**Comment by bpaulien** [comment]
I agree with @Deguello , If I put it in I intend to pay for it, and I don't want my budget reflecting that it's still available to be spent.  Let's say I have $100 for household stuff, and I order a crock pot for $50, but it won't ship (and therefore charge my card) until 2 weeks from now.  I want to put it in as a charge, dated 2 weeks from now, and I don't want it showing up as available in my budget, because then I still think I've got $100 to spend, when I really only have $50, it just hasn't been charged yet.

**Comment by MikeBishop** [comment]
And here we see the two camps.  🙂  Honestly, I'd like to see the future-dated transactions in the budget, but as a separate column.  Because they haven't happened yet, they could still be changed.  Future transactions are a plan, while present/past transactions are reality.
